src/connector/src/sink/remote.rs

```diff
@@ -228,14 +243,124 @@ async fn validate_remote_sink(param: &SinkParam) -> Result<()> {
     .map_err(|e| anyhow!("unable to validate: {:?}", e))?
 }
 
+enum StreamChunkConverter {
+    Es(EsStreamChunkConverter),
+    Other,
+}
```

Comment:
Not quite get the design, if the logic is specified to es, maybe we can implement it inside es sink.

Reply:
Like 

> In this PR, all processing is handled in rust side, including generating the doc id and processing the content body. On java side we just build and send request from the processed doc id and content body, so the pk index is not used on java side.

The goal is to make it easy to implement the struct  of es.